tests/ipam_then_pool/main.tf

```diff
@@ -0,0 +1,45 @@
+resource "random_string" "this" {
+  length  = 8
+  upper   = false
+  special = false
+  numeric = false
+}
+
+module "create_ipam" {
+  source = "../.."
+
+  vpc_ipam = {
+    ipam = {
+      name = "default_ipam"
+      operating_regions = [
+        { region_name = "us-east-1" },
+        { region_name = "us-west-1" }
+      ]
+      cascade     = true
+      description = random_string.this.result
+    }
+  }
+}
+
+module "create_pool" {
+  source = "../.."
+
+  vpc_ipam = {
+
+    pools = [
+      {
+        name                              = "pool_of_cidrs"
+        address_family                    = "ipv4"
+        allocation_default_netmask_length = "16"
+        allocation_min_netmask_length     = "16"
+        allocation_max_netmask_length     = "16"
+        description                       = random_string.this.result
+        ipam_scope_name                   = "private_default_scope"
```

Comment:
Ok, yeah, this is the test that is causing the empty tuple error. That's because the ipam resource was created in the other module call, so you can't use `ipam_scope_name` to look it up. you have to use `ipam_scope_id = module.create_ipam.ipam_out[0].private_default_scope_id`.